frontend/src/common/utils/api/readJson.ts

```diff
@@ -1,5 +1,21 @@
 import fs from "fs";
+import zlib from "zlib";
 
 export function readJson(path: string): any {
-  return JSON.parse(fs.readFileSync(path, "utf8"));
+  const data = fs.readFileSync(path);
+
+  // Check if the file is gzip compressed by examining the first two bytes (magic numbers)
+  const isGzipCompressed =
```

Comment:
Needed to add logic for decompressing gzip files - the allEnrichedGenes.json JSON got too big to store uncompressed in github.

Reply:
Hmm I wonder if enabling https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-git-large-file-storage would allow us to avoid this decompressing runtime cost?